### PR TITLE
Hookify TextField, Toast, and TopBar examples

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Convert examples to hooks

### WHAT is this pull request doing?

Converting TextField, Toast, and TopBar class-based examples to hooks

### STILL TO DO

I'm making quite a few pr's converting examples to hooks so I anticipate lots of `unreleased.md` conflicts, so I'll add the entry before I merge.

```
- Converted `TextField`, `Toast`, and `TopBar` examples to functional components ([#2135](https://github.com/Shopify/polaris-react/pull/2135))
```

### How to 🎩

Make sure the behavior of the example mirrors the style guide 

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

File: src/components/Toast/README.md [hunk 1]
```diff
@@ -166,32 +161,25 @@ Action should:
...
-    );
-  }
-
-  toggleToast = () => {
-    this.setState(({showToast}) => ({showToast: !showToast}));
-  };
+function ToastExample() {
+  const [active, setActive] = useState(false);
+
+  const toggleActive = useCallback(() => setActive(!active), [active]);
```
Comment (BPScott) [inline src/components/Toast/README.md]:
```suggestion
  const toggleActive = useCallback(() => setActive((state) => !state), []);
```

File: src/components/Toast/README.md [hunk 2]
```diff
@@ -202,45 +190,40 @@ class ToastExample extends React.Component {
...
-  };
-
-  toggleToast2 = () => {
-    this.setState(({showToast2}) => ({showToast2: !showToast2}));
-  };
+function MultipleToastExample() {
+  const [activeOne, setActiveOne] = useState(false);
+  const [activeTwo, setActiveTwo] = useState(false);
+
+  const toggleActiveOne = useCallback(() => setActiveOne(!activeOne), [
```
Comment (BPScott) [inline src/components/Toast/README.md]:
```
const toggleActiveOne = useCallback(() => setActiveOne((state) => !state), []);

const toggleActiveTwo = useCallback(() => setActiveTwo((state) => !state), []);
```

File: src/components/Toast/README.md [hunk 3]
```diff
@@ -251,36 +234,25 @@ class ToastExample extends React.Component {
...
-    );
-  }
-
-  toggleToast = () => {
-    this.setState(({showToast}) => ({showToast: !showToast}));
-  };
+function ToastWithCustomDurationExample() {
+  const [active, setActive] = useState(false);
+
+  const toggleActive = useCallback(() => setActive(!active), [active]);
```
Comment (BPScott) [inline src/components/Toast/README.md]:
```suggestion
const toggleActive = useCallback(() => setActive((state) => !state), []);
```

File: src/components/Toast/README.md [hunk 4]
```diff
@@ -291,40 +263,33 @@ class ToastExample extends React.Component {
...
-    );
-  }
-
-  toggleToast = () => {
-    this.setState(({showToast}) => ({showToast: !showToast}));
-  };
+function ToastWithActionExample() {
+  const [active, setActive] = useState(false);
+
+  const toggleActive = useCallback(() => setActive(!active), [active]);
```
Comment (BPScott) [inline src/components/Toast/README.md]:
```suggestion
const toggleActive = useCallback(() => setActive((state) => !state), []);
```

File: src/components/Toast/README.md [hunk 5]
```diff
@@ -377,32 +342,25 @@ Although error toast is still available and used in the system, we discourage it
...
-    );
-  }
-
-  toggleToast = () => {
-    this.setState(({showToast}) => ({showToast: !showToast}));
-  };
+function ErrorToastExample() {
+  const [active, setActive] = useState(false);
+
+  const toggleActive = useCallback(() => setActive(!active), [active]);
```
Comment (BPScott) [inline src/components/Toast/README.md]:
```suggestion
const toggleActive = useCallback(() => setActive((state) => !state), []);
```

File: src/components/TopBar/README.md [hunk 6]
```diff
@@ -146,136 +146,116 @@ A text field component that is tailor-made for a search use-case.
...
-  state = {
-    userMenuOpen: false,
-    searchActive: false,
-    searchText: '',
+function TopBarExample() {
+  const [isUserMenuOpen, setIsUserMenuOpen] = useState(false);
+  const [isSearchActive, setIsSearchActive] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  const toggleIsUserMenuOpen = useCallback(
```
Comment (BPScott) [inline src/components/TopBar/README.md]:
```
const toggleIsUserMenuOpen = useCallback(() => setIsUserMenuOpen((state) => !state), []);
```

File: src/components/TopBar/README.md [hunk 7]
```diff
@@ -284,138 +264,118 @@ class TopBarExample extends React.Component {
...
```
Comment (BPScott) [inline src/components/TopBar/README.md]:
```
const toggleIsUserMenuOpen = useCallback(() => setIsUserMenuOpen((state) => !state), []);
```

File: src/components/Toast/README.md (hunk 1 above)
Comment (LauraAubin) [reply src/components/Toast/README.md]: For my understanding, how does this not modify all of the values in state if we only want to change `active`?

Comment (AndrewMusgrave) [reply src/components/Toast/README.md]: For some reason, I assumed `set*` didn't accept functions or callbacks. Guess I was wrong about the former 😄 I'll comb through my other examples and update them where it's needed.

> For my understanding, how does this not modify all of the values in state if we only want to change active?

When using hooks your able to have a single state opposed to state objects in class-based components. If you do decide to use an object to hold state using `useState`, you won't be able to partially update state as you can in class base components.

```tsx
const [booleanValue, setBooleanValue] = useState(false)
const toggleBooleanValue = useCallback(() => setActive(booleanValue => !booleanValue), []);
```

Comment (BPScott) [reply src/components/Toast/README.md]: Crosslinking ya to the other PR where I mentioned this, which has some explanations and links to docs https://github.com/Shopify/polaris-react/pull/2126#discussion_r323986229

**Pull request merged**